File: app/models/comment.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.sql import func
from app.db.database import Base


class Comment(Base):
    __tablename__ = "comments"

    id = Column(Integer, primary_key=True, index=True)
    movie_id = Column(Integer, index=True, nullable=False)
    user_id = Column(Integer, index=True, nullable=False)
    parent_id = Column(Integer, nullable=True)
    body = Column(String, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

File: intelligent-movie-frontend/src/app/models/comment.ts
export interface CommentCreate {
  body: string;
  parent_id?: number;
}

export interface CommentRead {
  id: number;
  movie_id: number;
  user_id: number;
  parent_id?: number;
  body: string;
  created_at?: string;
  updated_at?: string;
  replies?: CommentRead[];
}
export interface Comment {
}
